pennylane/workflow/qnode.py

```diff
@@ -925,10 +926,19 @@ def construct(self, args, kwargs):  # pylint: disable=too-many-branches
         if old_interface == "auto":
             self.interface = qml.math.get_interface(*args, *list(kwargs.values()))
 
+        # Before constructing the tape, we pass the device to the
+        # debugger to ensure they are compatible if there are any
+        # breakpoints in the circuit
+        if PLDB.is_active_dev():
+            PLDB.reset_active_dev()
+
+        PLDB.add_device(self.device)

```

Comment:
# Suggestion
This works, although it could potentially make sense to just have the "setting" of an active device as a context manager (so that the caller doesn't need to "remember" to reset the active device). This is similar to the approach JAX uses.

Reply:
Added!